doc/source/dev/release.rst

```diff
@@ -0,0 +1,259 @@
+.. _dev_release:
+
+#######################
+The ODL release process
+#######################
+
+This document is intended to give precise instructions on the process of making a release.
+Its purpose is to avoid broken packages, broken documentation and many other things that can go wrong as a result of mistakes during the release process.
+Since this is not everyday work and may be done under the stress of a (self-imposed) deadline, it is clearly beneficial to have a checklist to hold on to.
+
+.. note::
+    The instructions in this document are tentative until tested in practice.
+    They are also written from the perspective of Linux and may need adaption for other platforms.
+
+
+.. _dev_rel_release_schedule:
+1. Agree on a release schedule
+------------------------------
+This involves the "what" and "when" of the release process and fixes a feature set that is supposed to be included in the new version.
+The steps are:
+
+- Open an issue on the issue tracker using the title **Release X.Y.Z** (insert numbers, of course).
+- Discuss and agree on a set of open PRs that should be merged before making a release.
+
+
+.. _dev_rel_master_ok:
+2. Make sure tests succeed and docs are built properly
+------------------------------------------------------
+When all required PRs are merged, ensure that the latest ``master`` branch is sane. Travis CI checks every PR, but certain things like CUDA cannot be tested there and must therefore undergo tests on a local machine, for at least Python 2.7 and one version of Python 3.
+
+- Make a new test conda environment and install all dependencies:
+
+  .. code-block:: bash
+
+    conda create -n release36 python=3.6 nomkl numpy scipy future matplotlib pytest
+    source activate release36
+    pip install pyfftw pywavelets
+    cd /path/to/odl_repo
+    git checkout master
+    git pull
+    pip install -e .
+
+- Run the tests with ``pytest``, including doctests, examples documentation and large-scale tests:
+
+  .. code-block:: bash
+
+    pytest --doctest-modules --examples --doctest-doc --largescale
+
+- Do the same for Python 2.7.
+- Make sure the tests also run on the platforms you're currently *not* testing on.
+  Ask a buddy maintainer if necessary.
+- Build the documentation.
+  This requires ``sphinx`` and the ``sphinxext`` submodule:
+
+  .. code-block:: bash
+
+    conda install sphinx sphinx_rtd_theme
+    git submodule update --init --recursive
+    cd doc && make clean
+    cd source && python generate_doc.py
+    cd ..
+    make html 2>&1 |\
+      grep -E "SEVERE|ERROR|WARNING" |\
+      grep -E -v "more than one target found for|__eq__|document isn't included in any toctree"
+
+  The last command builds the documentation and filters from the output all irrelevant warnings, letting through only the "proper" warnings and errors.
```

Comment:
Make an issue about fixing these warnings and link here, its a bit sad right now.